pkg/k8s/node.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"net"
+
+	"github.com/cilium/cilium/pkg/node"
+
+	log "github.com/Sirupsen/logrus"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+func ParseNode(node *v1.Node) *node.Node {
+	addrs := []node.NodeAddress{}
+	for _, addr := range node.Status.Addresses {
+		// We only care about this address types,
+		// we ignore all other types.
+		switch addr.Type {
+		case v1.NodeLegacyHostIP, v1.NodeInternalIP, v1.NodeExternalIP:
+		default:
+			continue
+		}
+		ip := net.ParseIP(addr.Address)
+		if ip == nil {
+			log.Debugf("Ignoring invalid IP address %s of type %s", addr.Address, addr.Type)
+			continue
+		}
+		na := node.NodeAddress{
+			AddressType: addr.Type,
+			IP:          ip,
+		}
+		addrs = append(addrs, na)
+	}
+
+	var endpointsCIDR *net.IPNet
+	if len(node.Spec.PodCIDR) != 0 {
+		var err error
+		_, endpointsCIDR, err = net.ParseCIDR(node.Spec.PodCIDR)
+		if err != nil {
+			log.Warningf("Invalid PodCIDR %s: %s", node.Spec.PodCIDR, err)
```

Comment:
`"k8s: Invalid PodCIDR value '%s' for node %s: %s", node.Spec.PodCIDR, node.Name, err)`